static/src/javascripts/projects/common/modules/experiments/tests/weekend-reading-promo.js

```diff
@@ -68,10 +87,12 @@ define([
             {
                 id: 'show',
                 test: function () {
-                    var messageText = 'Get the best reads of the week in your inbox every Saturday';
-                    var linkText = 'Sign up';
-                    var linkHref = '/survey/weekendreading?CMP=SnapBanner';
-                    renderDigestSnap(messageText, linkText, linkHref, 'weekend-reading');
+                    mediator.on('membership-message:complete', function () {
+                        var messageText = 'Get the best reads of the week in your inbox every Saturday';
+                        var linkText = 'Sign up';
+                        var linkHref = '/signup/weekendreading?CMP=SnapBanner';
+                        renderWeekendReadingSnap(messageText, linkText, linkHref, 'weekend-reading');
+                    });
                 }
```

Comment:
As discussed, in cases like these where you only want to call a function after multiple events have fired, [`_.after()`](https://lodash.com/docs/4.15.0#after) can be useful